docs/radio.rst

```diff
@@ -48,8 +48,13 @@ Functions
 
 .. py:function:: on()
 
-    Turns the radio on. This needs to be explicitly called since the radio
-    draws power and takes up memory that you may otherwise need.
+    Turns the radio on.
```

Comment:
Reading this now, I realise we did not consider the point that enabling the radio takes up RAM (as well as power), which is in very short supply on micro:bit v1.

Turning the radio on at import may break some code and be impossible to fix.  It could be the case that the user's program is very tight on memory and they do something like this:
```python
import radio
radio.config(length=10, queue=1)
radio.on()  # this is where the memory is allocated for radio buffers
```
In the approach now where it's turned on at import, it will allocate the default buffer space, which is length=32 and queue=3.  That default allocates (32+6)*(3+1) = 152 bytes.  Then in the above code it will free this and reallocate to the smaller size, which is (10+6)*(1+1) = 32 bytes.  The import may fail if the micro:bit doesn't have much free RAM left.

I'm not sure how much of an issue this is but it needs to be considered.

Reply:
Yeah, printing the free memory size before and after `radio.on()` aligns to your calculation and shows a decrease of about 160 bytes.

You are right that this can cause issues, but I think most users will likely configure the radio very soon after import, as your example shows.

In those cases, hopefully, there is more than 120 bytes available at that point, as you'd expect the main.py file to contain at least some code to use memory as well.

We'll release this as a beta release in the Python Editor beta, and that way we can consider revert it if there are issues before the main release.

Reply:
I think it's unlikely to cause an issue using a little more memory like this.  I doubt many users realise they can reclaim a few 10's of bytes of RAM by reconfiguring the radio like this, so it's probably never done in practice (what's more likely is that they *increase* the length/queue size to get better radio performance).